system-addon/content-src/components/Search/Search.jsx

```diff
@@ -0,0 +1,34 @@
+"use strict";
+const React = require("react");
+const {connect} = require("react-redux");
+const {actionTypes, actionCreators} = require("common/Actions.jsm");
+
+const Search = React.createClass({
+  getInitialState() {
+    return {
+      searchString: ''
+    };
+  },
+  performSearch(options) {
+    let searchData = {
+      engineName: options.engineName,
+      searchString: options.searchString,
+      searchPurpose: "d",
```

Comment:
This was copied over, but should we be putting something more useful for `purpose`? It looks like existing Firefox usage might be "newtab" and "homepage"